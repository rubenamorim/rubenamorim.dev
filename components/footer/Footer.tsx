import { Email, LinkedIn, Github, Twitter } from 'components/icons';
import ExternalLink from 'components/external-link/ExternalLink';

const LINKS = [
    {
        id: 'email',
        url: 'mailto:me@rubenamorim.dev?body=Hey%20%F0%9F%91%8B%0D%0A',
        title: 'Email',
        label: 'Send an email to Ruben',
        icon: Email,
    },
    {
        id: 'linkedin',
        url: 'https://www.linkedin.com/in/rubenamorim/',
        title: 'LinkedIn',
        label: "Open Ruben's LinkedIn profile",
        icon: LinkedIn,
    },
    {
        id: 'twitter',
        url: 'https://twitter.com/rubentamorim',
        title: 'Twitter',
        label: "Open Ruben's Twitter profile",
        icon: Twitter,
    },
    {
        id: 'github',
        url: 'https://github.com/rubenamorim',
        title: 'Github',
        label: "Open Ruben's Github profile",
        icon: Github,
    },
];

const Footer: React.FC = () => {
    return (
        <footer className="h-28 flex justify-center items-center md:h-40">
            <nav aria-label="Social navigation">
                <ul className="flex flex-row">
                    {LINKS.map(({ id, label, url, title, icon: Icon }) => (
                        <li key={id} className="mr-14 last:mr-0">
                            <ExternalLink
                                href={url}
                                aria-label={label}
                                title={title}
                                className="block text-4xl md:text-5xl transition-transform hover:scale-110 focus:scale-110"
                            >
                                <Icon />
                            </ExternalLink>
                        </li>
                    ))}
                </ul>
            </nav>
        </footer>
    );
};

export default Footer;
