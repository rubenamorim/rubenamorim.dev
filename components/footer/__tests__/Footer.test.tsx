import { render, screen } from '@testing-library/react';

import Footer from '../Footer';

describe('Footer', () => {
    it('should render correctly', () => {
        const { container } = render(<Footer />);

        expect(
            screen.getByLabelText('Send an email to Ruben')
        ).toBeInTheDocument();
        expect(
            screen.getByLabelText("Open Ruben's LinkedIn profile")
        ).toBeInTheDocument();
        expect(
            screen.getByLabelText("Open Ruben's Twitter profile")
        ).toBeInTheDocument();
        expect(
            screen.getByLabelText("Open Ruben's Github profile")
        ).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
