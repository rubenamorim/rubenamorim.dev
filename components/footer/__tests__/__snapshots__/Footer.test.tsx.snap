// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<div>
  <footer
    class="h-28 flex justify-center items-center md:h-40"
  >
    <nav
      aria-label="Social navigation"
    >
      <ul
        class="flex flex-row"
      >
        <li
          class="mr-14 last:mr-0"
        >
          <a
            aria-label="Send an email to Ruben"
            class="block text-4xl md:text-5xl transform transition-transform hover:scale-110 focus:scale-110"
            href="mailto:me@rubenamorim.dev?body=Hey%20%F0%9F%91%8B%0D%0A"
            rel="nofollow noopener noreferrer"
            target="_blank"
            title="Email"
          >
            <svg
              height="1em"
              viewBox="0 0 30 30"
              width="1em"
            >
              <path
                d="M30 0l-7.5 27.5-10.161-9.049L22.09 8.16 9.02 17.192 0 15 30 0zM11.25 20.835V30l4.072-5.539-4.072-3.626z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
        <li
          class="mr-14 last:mr-0"
        >
          <a
            aria-label="Open Ruben's LinkedIn profile"
            class="block text-4xl md:text-5xl transform transition-transform hover:scale-110 focus:scale-110"
            href="https://www.linkedin.com/in/rubenamorim/"
            rel="nofollow noopener noreferrer"
            target="_blank"
            title="LinkedIn"
          >
            <svg
              height="1em"
              viewBox="0 0 30 30"
              width="1em"
            >
              <path
                d="M25.605 0H4.395A4.4 4.4 0 000 4.395v21.21A4.4 4.4 0 004.395 30h21.21A4.4 4.4 0 0030 25.605V4.395A4.4 4.4 0 0025.605 0zm-15 23.79H7.09V11.483h3.515V23.79zm0-14.063H7.09V6.21h3.515v3.516zM22.91 23.789h-3.515v-7.031A1.76 1.76 0 0017.637 15a1.76 1.76 0 00-1.758 1.758v7.031h-3.516V11.484h3.516v.663c.92-.286 1.52-.663 2.637-.663 2.384.003 4.394 2.142 4.394 4.67v7.635z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
        <li
          class="mr-14 last:mr-0"
        >
          <a
            aria-label="Open Ruben's Twitter profile"
            class="block text-4xl md:text-5xl transform transition-transform hover:scale-110 focus:scale-110"
            href="https://twitter.com/rubentamorim"
            rel="nofollow noopener noreferrer"
            target="_blank"
            title="Twitter"
          >
            <svg
              height="1em"
              viewBox="0 0 36.92 30"
              width="1.23em"
            >
              <path
                d="M36.923 3.552a15.782 15.782 0 01-4.361 1.195 7.526 7.526 0 003.33-4.184 15.126 15.126 0 01-4.8 1.832A7.585 7.585 0 0025.561 0c-4.19 0-7.564 3.402-7.564 7.572 0 .6.05 1.176.175 1.726C11.88 8.99 6.312 5.975 2.571 1.38a7.62 7.62 0 00-1.036 3.826 7.58 7.58 0 003.362 6.29 7.475 7.475 0 01-3.42-.931v.083c0 3.678 2.624 6.733 6.065 7.437a7.558 7.558 0 01-1.985.25c-.485 0-.974-.028-1.433-.13.98 2.998 3.764 5.202 7.073 5.274a15.21 15.21 0 01-9.385 3.228c-.621 0-1.217-.028-1.812-.104C3.353 28.765 7.327 30 11.612 30c13.93 0 21.545-11.538 21.545-21.54 0-.335-.012-.658-.028-.978 1.503-1.067 2.765-2.398 3.794-3.93z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
        <li
          class="mr-14 last:mr-0"
        >
          <a
            aria-label="Open Ruben's Github profile"
            class="block text-4xl md:text-5xl transform transition-transform hover:scale-110 focus:scale-110"
            href="https://github.com/rubenamorim"
            rel="nofollow noopener noreferrer"
            target="_blank"
            title="Github"
          >
            <svg
              height="1em"
              viewBox="0 0 30 30"
              width="1em"
            >
              <path
                d="M15 0C6.718 0 0 6.716 0 15c0 6.628 4.298 12.25 10.259 14.234.748.138.991-.327.991-.722V25.72c-4.173.908-5.041-1.77-5.041-1.77-.683-1.734-1.666-2.195-1.666-2.195-1.362-.931.103-.911.103-.911 1.506.105 2.299 1.546 2.299 1.546 1.338 2.293 3.509 1.63 4.365 1.246.134-.968.522-1.631.952-2.005-3.33-.381-6.833-1.667-6.833-7.414 0-1.638.586-2.976 1.545-4.026-.155-.379-.669-1.905.146-3.97 0 0 1.26-.402 4.126 1.538A14.385 14.385 0 0115 7.254c1.275.006 2.559.172 3.758.505 2.863-1.94 4.12-1.538 4.12-1.538.817 2.066.303 3.593.148 3.97.963 1.05 1.544 2.389 1.544 4.026 0 5.762-3.509 7.03-6.849 7.402.538.465 1.029 1.377 1.029 2.777v4.116c0 .4.24.868 1.001.72C25.707 27.247 30 21.626 30 15c0-8.284-6.716-15-15-15z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;
